File: src/com/eagle/programmar/Django/Controls/Django_IfControl.java
// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Apr 15, 2014

package com.eagle.programmar.Django.Controls;

import com.eagle.programmar.Django.Django_Element;
import com.eagle.programmar.Django.Django_Expression;
import com.eagle.programmar.Django.Terminals.Django_Keyword;
import com.eagle.programmar.HTML.Terminals.HTML_Punctuation;
import com.eagle.tokens.TokenList;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationHyphen;

public class Django_IfControl extends TokenSequence
{
	public @S(10) Django_Keyword IF = new Django_Keyword("if");
	public @S(20) Django_Expression expr;
	public @S(30) @OPT PunctuationHyphen dash2;
	public @S(40) HTML_Punctuation percentBrace1 = new HTML_Punctuation("%}");

	public @S(50) TokenList<Django_Element> html;

	public @S(60) @OPT Django_IfElseControl ifElse;

	public @S(70) HTML_Punctuation bracePercent2 = new HTML_Punctuation("{%");
	public @S(80) @OPT PunctuationHyphen dash3;
	public @S(90) Django_Keyword ENDIF = new Django_Keyword("endif");

	public static class Django_IfElseControl extends TokenSequence
	{
		public @S(10) HTML_Punctuation bracePercent = new HTML_Punctuation("{%");
		public @S(20) @OPT PunctuationHyphen dash1;
		public @S(30) Django_Keyword ELSE = new Django_Keyword("else");
		public @S(40) @OPT PunctuationHyphen dash2;
		public @S(50) HTML_Punctuation percentBrace = new HTML_Punctuation("%}");

		public @S(60) TokenList<Django_Element> html;
	}
}

File: csrc/com/eagle/programmar/Django/Controls/Django_IfControl.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Apr 15, 2014

namespace com.eagle.programmar.Django.Controls
{
	using Django_Element = com.eagle.programmar.Django.Django_Element;
	using Django_Expression = com.eagle.programmar.Django.Django_Expression;
	using Django_Keyword = com.eagle.programmar.Django.Terminals.Django_Keyword;
	using HTML_Punctuation = com.eagle.programmar.HTML.Terminals.HTML_Punctuation;
	using TokenList = com.eagle.tokens.TokenList;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationHyphen = com.eagle.tokens.punctuation.PunctuationHyphen;

	public class Django_IfControl : TokenSequence
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.Django.Terminals.Django_Keyword IF = new com.eagle.programmar.Django.Terminals.Django_Keyword("if");
		public Django_Keyword IF = new Django_Keyword("if");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.Django.Django_Expression expr;
		public Django_Expression expr;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) @OPT PunctuationHyphen dash2;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) com.eagle.programmar.HTML.Terminals.HTML_Punctuation percentBrace1 = new com.eagle.programmar.HTML.Terminals.HTML_Punctuation("%}");
		public HTML_Punctuation percentBrace1 = new HTML_Punctuation("%}");

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) com.eagle.tokens.TokenList<com.eagle.programmar.Django.Django_Element> html;
		public TokenList<Django_Element> html;

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(60) @OPT Django_IfElseControl ifElse;
		public  OPT;

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(70) com.eagle.programmar.HTML.Terminals.HTML_Punctuation bracePercent2 = new com.eagle.programmar.HTML.Terminals.HTML_Punctuation("{%");
		public HTML_Punctuation bracePercent2 = new HTML_Punctuation("{%");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(80) @OPT PunctuationHyphen dash3;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(90) com.eagle.programmar.Django.Terminals.Django_Keyword ENDIF = new com.eagle.programmar.Django.Terminals.Django_Keyword("endif");
		public Django_Keyword ENDIF = new Django_Keyword("endif");

		public class Django_IfElseControl : TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.HTML.Terminals.HTML_Punctuation bracePercent = new com.eagle.programmar.HTML.Terminals.HTML_Punctuation("{%");
			public HTML_Punctuation bracePercent = new HTML_Punctuation("{%");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT PunctuationHyphen dash1;
			public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.Django.Terminals.Django_Keyword ELSE = new com.eagle.programmar.Django.Terminals.Django_Keyword("else");
			public Django_Keyword ELSE = new Django_Keyword("else");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) @OPT PunctuationHyphen dash2;
			public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) com.eagle.programmar.HTML.Terminals.HTML_Punctuation percentBrace = new com.eagle.programmar.HTML.Terminals.HTML_Punctuation("%}");
			public HTML_Punctuation percentBrace = new HTML_Punctuation("%}");

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(60) com.eagle.tokens.TokenList<com.eagle.programmar.Django.Django_Element> html;
			public TokenList<Django_Element> html;
		}
	}

}
